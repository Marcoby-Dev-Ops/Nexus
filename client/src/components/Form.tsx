import React from 'react';
import PropTypes from 'prop-types';

/**
 * @interface FormProps
 * @description Props for the Form component.
 */
export interface FormProps extends React.FormHTMLAttributes<HTMLFormElement> {
  onSubmit: (event: React.FormEvent<HTMLFormElement>) => void;
  children: React.ReactNode;
  className?: string;
}

/**
 * @name Form
 * @description A form component for grouping form elements with validation.
 * @param {FormProps} props - The props for the component.
 * @returns {JSX.Element} The rendered Form component.
 */
export const Form: React.FC<FormProps> = React.memo(({ onSubmit, children, className = '', ...props }) => (
  <form
    className={`space-y-4 ${className}`}
    onSubmit={onSubmit}
    noValidate
    {...props}
  >
    {children}
  </form>
));

Form.displayName = 'Form';

Form.propTypes = {
  onSubmit: PropTypes.func.isRequired,
  children: PropTypes.node.isRequired,
  className: PropTypes.string,
};

export default Form; 