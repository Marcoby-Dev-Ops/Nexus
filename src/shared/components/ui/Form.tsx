import React from 'react';
import { cn } from '@/shared/utils/styles';

export interface FormProps extends React.FormHTMLAttributes<HTMLFormElement> {
  children: React.ReactNode;
}

export interface FormFieldProps {
  label?: string;
  error?: string;
  required?: boolean;
  children: React.ReactNode;
  className?: string;
}

/**
 * Simple form components for 1.0
 */
const Form: React.FC<FormProps> = ({ children, className, ...props }) => {
  return (
    <form className={cn('space-y-4', className)} {...props}>
      {children}
    </form>
  );
};

export const FormField: React.FC<FormFieldProps> = ({ 
  label, 
  error, 
  required, 
  children, 
  className 
}) => {
  return (
    <div className={cn('space-y-2', className)}>
      {label && (
        <label className="text-sm font-medium leading-none peer-disabled:cursor-not-allowed peer-disabled:opacity-70">
          {label}
          {required && <span className="text-destructive ml-1">*</span>}
        </label>
      )}
      {children}
      {error && (
        <p className="text-sm text-destructive">{error}</p>
      )}
    </div>
  );
};

export default Form; 