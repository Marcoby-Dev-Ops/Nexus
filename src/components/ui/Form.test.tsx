/**
 * @file Form.test.tsx
 * @description Unit and snapshot tests for the Form component.
 */
import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom';
import Form from './Form';

describe('Form', () => {
  it('renders children', () => {
    render(<Form>Form Content</Form>);
    expect(screen.getByText('Form Content')).toBeInTheDocument();
  });

  it('calls onSubmit when submitted', () => {
    const handleSubmit = jest.fn((e) => e.preventDefault());
    render(<Form onSubmit={handleSubmit}><button type="submit">Submit</button></Form>);
    fireEvent.click(screen.getByText('Submit'));
    expect(handleSubmit).toHaveBeenCalled();
  });

  it('matches snapshot', () => {
    const { asFragment } = render(<Form>Snapshot Form</Form>);
    expect(asFragment()).toMatchSnapshot();
  });
}); 