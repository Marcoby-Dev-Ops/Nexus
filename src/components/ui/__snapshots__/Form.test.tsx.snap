// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form matches snapshot 1`] = `
<DocumentFragment>
  <form
    class="space-y-4"
  >
    Snapshot Form
  </form>
</DocumentFragment>
`;
